internal/bsdiff/bsdiff.go

```
@@ -0,0 +1,222 @@
+}
+
+// DiffReference is bsdiff 4.3 exactly: it finds matches through a suffix array
+// of old and emits the same patch as the C tool, several times slower than Diff.
+func DiffReference(old, new []byte) ([]byte, error) {
+	if len(old) > maxSize || len(new) > maxSize {
+		return nil, errTooLarge
+	}
+	I := make([]int32, len(old)+1)
+	V := make([]int32, len(old)+1)
+	qsufsort(I, V, old)
+	V = nil
```

golangci-lint (2.13.2)

[error] ineffectual assignment to V

(ineffassign)